python/private/pypi/pip_repository_attrs.bzl

```diff
@@ -21,6 +21,21 @@ repositories."""
 load(":attrs.bzl", COMMON_ATTRS = "ATTRS")
 
 ATTRS = {
+    "extra_pip_args_by_platform": attr.string_list_dict(
+        doc = """Similar to `extra_pip_args`.
+
+Extra arguments to pass on to pip on the assoicated platform specifiers. Args must not contain spaces.
```

Comment:
Could you please update https://github.com/bazel-contrib/rules_python/blob/main/docs/pypi-dependencies.md?plain=1#L320 section on how this feature could be used?